<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Widget</class>
 <widget class="QWidget" name="Widget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>768</width>
    <height>925</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>hw3_mask</string>
  </property>
  <property name="autoFillBackground">
   <bool>false</bool>
  </property>
  <widget class="QListWidget" name="listWidget_ImageList">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>80</y>
     <width>331</width>
     <height>351</height>
    </rect>
   </property>
  </widget>
  <widget class="QPushButton" name="pushButton_ChooseFolder">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>20</y>
     <width>220</width>
     <height>30</height>
    </rect>
   </property>
   <property name="text">
    <string>Choose your image folder.</string>
   </property>
  </widget>
  <widget class="QGroupBox" name="groupBox">
   <property name="geometry">
    <rect>
     <x>370</x>
     <y>30</y>
     <width>380</width>
     <height>400</height>
    </rect>
   </property>
   <property name="title">
    <string>Original Image</string>
   </property>
   <widget class="QLabel" name="label_ori">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>30</y>
      <width>360</width>
      <height>360</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
  </widget>
  <widget class="QLabel" name="label_ImageList">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>60</y>
     <width>220</width>
     <height>20</height>
    </rect>
   </property>
   <property name="text">
    <string>Image list:</string>
   </property>
  </widget>
  <widget class="QGroupBox" name="groupBox_2">
   <property name="geometry">
    <rect>
     <x>370</x>
     <y>450</y>
     <width>380</width>
     <height>400</height>
    </rect>
   </property>
   <property name="title">
    <string>Image(after)</string>
   </property>
   <widget class="QLabel" name="label_aft">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>30</y>
      <width>360</width>
      <height>360</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
  </widget>
  <widget class="QGroupBox" name="groupBox_3">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>450</y>
     <width>331</width>
     <height>461</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <weight>50</weight>
     <bold>false</bold>
    </font>
   </property>
   <property name="title">
    <string>Set your mask.</string>
   </property>
   <widget class="QLabel" name="label_step1">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>30</y>
      <width>201</width>
      <height>21</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <weight>75</weight>
      <bold>true</bold>
      <underline>true</underline>
     </font>
    </property>
    <property name="text">
     <string>Step1: Adjust mask's size.</string>
    </property>
   </widget>
   <widget class="QSlider" name="horizontalSlider_MaskSize">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>55</y>
      <width>151</width>
      <height>16</height>
     </rect>
    </property>
    <property name="minimum">
     <number>3</number>
    </property>
    <property name="maximum">
     <number>50</number>
    </property>
    <property name="singleStep">
     <number>2</number>
    </property>
    <property name="pageStep">
     <number>2</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QLabel" name="label_step2">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>70</y>
      <width>201</width>
      <height>21</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <weight>75</weight>
      <italic>false</italic>
      <bold>true</bold>
      <underline>true</underline>
     </font>
    </property>
    <property name="text">
     <string>Step2: Create a mask.</string>
    </property>
   </widget>
   <widget class="QTabWidget" name="tabWidget_CreateMask">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>100</y>
      <width>300</width>
      <height>291</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>3</number>
    </property>
    <widget class="QWidget" name="tab_Linear">
     <attribute name="title">
      <string>Linear</string>
     </attribute>
     <widget class="QRadioButton" name="radioButton_GKernal">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>30</y>
        <width>170</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>Gaussion kernal</string>
      </property>
     </widget>
     <widget class="QRadioButton" name="radioButton_BoxKernal">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>170</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>Box kernel</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_Nonlinear">
     <attribute name="title">
      <string>Non-linear</string>
     </attribute>
     <widget class="QRadioButton" name="radioButton_MinFilter">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>50</y>
        <width>171</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Min filter</string>
      </property>
     </widget>
     <widget class="QRadioButton" name="radioButton_MaxFilter">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>30</y>
        <width>171</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Max filter</string>
      </property>
     </widget>
     <widget class="QRadioButton" name="radioButton_MedFilter">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>171</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Median filter</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_Custom">
     <attribute name="title">
      <string>Custom</string>
     </attribute>
     <widget class="QLabel" name="label_8">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>10</y>
        <width>200</width>
        <height>20</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <weight>75</weight>
        <bold>true</bold>
        <underline>true</underline>
       </font>
      </property>
      <property name="text">
       <string>Type your custom mask.</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_9">
      <property name="geometry">
       <rect>
        <x>25</x>
        <y>30</y>
        <width>200</width>
        <height>17</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>Example:</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_10">
      <property name="geometry">
       <rect>
        <x>60</x>
        <y>70</y>
        <width>91</width>
        <height>51</height>
       </rect>
      </property>
      <property name="text">
       <string>1 2 3
4 5 6          -&gt; 
7 8 9</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_11">
      <property name="geometry">
       <rect>
        <x>180</x>
        <y>70</y>
        <width>51</width>
        <height>51</height>
       </rect>
      </property>
      <property name="text">
       <string>1,2,3
4,5,6
7,8,9</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_12">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>50</y>
        <width>101</width>
        <height>17</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>Mask type:</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_13">
      <property name="geometry">
       <rect>
        <x>150</x>
        <y>50</y>
        <width>101</width>
        <height>17</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>Input type:</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_14">
      <property name="geometry">
       <rect>
        <x>25</x>
        <y>130</y>
        <width>200</width>
        <height>17</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>Input:</string>
      </property>
     </widget>
     <widget class="QTableWidget" name="tableWidget">
      <property name="geometry">
       <rect>
        <x>35</x>
        <y>150</y>
        <width>131</width>
        <height>101</height>
       </rect>
      </property>
      <property name="rowCount">
       <number>3</number>
      </property>
      <property name="columnCount">
       <number>3</number>
      </property>
      <attribute name="horizontalHeaderDefaultSectionSize">
       <number>20</number>
      </attribute>
      <attribute name="verticalHeaderDefaultSectionSize">
       <number>20</number>
      </attribute>
      <row/>
      <row/>
      <row/>
      <column/>
      <column/>
      <column/>
     </widget>
    </widget>
    <widget class="QWidget" name="tab">
     <attribute name="title">
      <string>Edge</string>
     </attribute>
     <widget class="QRadioButton" name="radioButton">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>141</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>LoG, threshold = </string>
      </property>
     </widget>
     <widget class="QSlider" name="horizontalSlider">
      <property name="geometry">
       <rect>
        <x>39</x>
        <y>40</y>
        <width>231</width>
        <height>20</height>
       </rect>
      </property>
      <property name="maximum">
       <number>5000</number>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
     <widget class="QRadioButton" name="radioButton_2">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>60</y>
        <width>141</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>Sobel</string>
      </property>
     </widget>
     <widget class="QLCDNumber" name="lcdNumber_2">
      <property name="geometry">
       <rect>
        <x>150</x>
        <y>10</y>
        <width>61</width>
        <height>23</height>
       </rect>
      </property>
      <property name="digitCount">
       <number>4</number>
      </property>
     </widget>
    </widget>
   </widget>
   <widget class="QLabel" name="label_step3">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>400</y>
      <width>201</width>
      <height>21</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <weight>75</weight>
      <italic>false</italic>
      <bold>true</bold>
      <underline>true</underline>
     </font>
    </property>
    <property name="text">
     <string>Step3: Run!</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_Run">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>425</y>
      <width>89</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>Run!</string>
    </property>
   </widget>
   <widget class="QLCDNumber" name="lcdNumber">
    <property name="geometry">
     <rect>
      <x>163</x>
      <y>52</y>
      <width>31</width>
      <height>21</height>
     </rect>
    </property>
    <property name="digitCount">
     <number>2</number>
    </property>
    <property name="value" stdset="0">
     <double>3.000000000000000</double>
    </property>
   </widget>
  </widget>
  <widget class="QProgressBar" name="progressBar">
   <property name="geometry">
    <rect>
     <x>370</x>
     <y>880</y>
     <width>381</width>
     <height>21</height>
    </rect>
   </property>
   <property name="value">
    <number>0</number>
   </property>
  </widget>
  <widget class="QLabel" name="label">
   <property name="geometry">
    <rect>
     <x>620</x>
     <y>900</y>
     <width>80</width>
     <height>20</height>
    </rect>
   </property>
   <property name="text">
    <string>0</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
   </property>
  </widget>
  <widget class="QLabel" name="label_2">
   <property name="geometry">
    <rect>
     <x>710</x>
     <y>900</y>
     <width>20</width>
     <height>20</height>
    </rect>
   </property>
   <property name="text">
    <string>ms</string>
   </property>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
